torch/_inductor/ir.py

```diff
@@ -4012,15 +4069,14 @@ def create(cls, inputs, dim):
                 and is_gpu(inputs[i].get_device().type)
                 and not is_dynamic(input_buffer)
             ):
-                buffer_names.append(input_buffer.get_name())
+                op_names.append(input_buffer.get_operation_name())
 
-        if len(buffer_names) > 1 and V.graph.has_feature(
-            device, BackendFeature.FOREACH
-        ):
-            V.graph.register_list(buffer_names)
+        if len(op_names) > 1 and V.graph.has_feature(device, BackendFeature.FOREACH):
+            V.graph.register_list(op_names)
 
         concat_kernel.name = V.graph.register_buffer(concat_kernel)
         concat_kernel.inputs = cls.unwrap_storage(concat_kernel.inputs)
+        V.graph.register_operation(concat_kernel)
```

Comment:
Any chance we can lift this to `OperationBuffer.__init__` as `V.graph.register_operation(self)` and then tagging manually the ones that have multiple outputs? It'd be nice for correctness not having to sprinkle all these in so many places.

Reply:
The problem is that sometimes we create IR nodes just to access their properties and then discard them, in which case we don't want them to be registered.

Reply:
Does that happen very often? If not, perhaps we want to have some context manager `no_register_op` to control this behaviour locally in those cases. This way we know for sure we never miss a register operation.
I'm not hell-bent on this though. If you think this is more of a hassle than the benefit it brings then disregard.

Reply:
Not a huge fan of using a context manager for this.  I plan did plan on adding a `register: bool` keyword argument to some operation subclasses constructors though.

Reply:
yeah. The thing with the kwarg is that you have to push it all the way to the base class, which is a bit annoying, but yep, not a big deal either way.